packages/components/src/query-controls/stories/index.tsx

```diff
@@ -0,0 +1,168 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import QueryControls from '..';
+import type { Category, QueryControlsProps } from '../types';
+
+const meta: ComponentMeta< typeof QueryControls > = {
+	title: 'Components/QueryControls',
+	component: QueryControls,
+	argTypes: {
+		numberOfItems: { control: { type: null } },
+		order: { control: { type: null } },
+		orderBy: { control: { type: null } },
+		selectedAuthorId: { control: { type: null } },
+		selectedCategories: { control: { type: null } },
+		selectedCategoryId: { control: { type: null } },
```

Comment:
If it's alright, I disabled the controls for values that the user can now change via the component.

It looks like other components do this.

For example, the RangeControl [disables](https://github.com/WordPress/gutenberg/blob/a35b3f245a9672e2a276d8c4c1785f42b3342f84/packages/components/src/range-control/stories/index.tsx#L46) the control for the `value` prop, as the user can change the value via the component.